src/nvim/mbyte.c

```diff
@@ -1320,40 +1321,10 @@ int utf_fold(int a)
   return utf_convert(a, foldCase, ARRAY_SIZE(foldCase));
 }
 
-// Vim's own character class functions.  These exist because many library
-// islower()/toupper() etc. do not work properly: they crash when used with
-// invalid values or can't handle latin1 when the locale is C.
-// Speed is most important here.
-
-// Note: UnicodeData.txt does not define U+1E9E as being the corresponding upper
-// case letter for U+00DF (ß), however it is part of the toLower table
-
-/// Return the upper-case equivalent of "a", which is a UCS-4 character.  Use
-/// simple case folding.
-int mb_toupper(int a)
-{
-  // If 'casemap' contains "keepascii" use ASCII style toupper().
-  if (a < 128 && (cmp_flags & CMP_KEEPASCII)) {
-    return TOUPPER_ASC(a);
-  }
-
-  if (!(cmp_flags & CMP_INTERNAL)) {
-    return (int)towupper((wint_t)a);
-  }
-
-  // For characters below 128 use locale sensitive toupper().
-  if (a < 128) {
-    return TOUPPER_LOC(a);
-  }
-
-  // For any other characters use the above mapping table.
-  return utf_convert(a, toUpper, ARRAY_SIZE(toUpper));
-}
-
 bool mb_islower(int a)
 {
   // German sharp s is lower case but has no upper case equivalent.
-  return (mb_toupper(a) != a) || a == 0xdf;
+  return (utf8proc_toupper(a) != a) || a == 0xdf;
```

Comment:
German sharp s upper case at home: ẞ

I.e., the whole rationale for this custom function is wrong now -- uppercase ß was added in 2017 -- so we should just use `utf8proc_islower`.

Reply:
I think I'll leave this alone in this PR. The reason is that what is considered uppercase/lowercase depends on user options, and I don't want to mess with them yet. I think it'd be best if we isolated the removal later.

Reply:
This one doesn't, though -- it's hardcoded, and wrong. I'm just saying keeping _this_ special behavior is not worth it and shouldn't block a change you would otherwise want to make (even if you want to keep the `keepascii` option, which I agree makes sense).

Reply:
Oh, you mean remove just the `|| a == 0xdf` part? Yeah, we could do that. Wonder if we should send PR to vim, this sounds relevant for them as well.

Reply:
I don't think they have the tables for that ;)